syntax = "proto2";

import "nanopb.proto";

package protocol;

message Report {
  optional int64 timestamp = 1;
  required State state = 2;
}

message State {
  required bool started = 1;
  required bool armed = 2;
  required bool modem_sleeping = 3;
  required bool reconnecting = 4;
  required bool doors_open = 5;
}

message RemoteSignal {
  required string client_id = 1 [(nanopb).max_length = 4];
  required bytes payload = 2 [(nanopb).max_size = 14];
  required bytes auth_tag = 3 [(nanopb).max_size = 8];
}

message RemoteSignalPayload {
  required uint32 code = 1;
  required bytes random = 2 [(nanopb).max_size = 10];
}

message RemoteSignalResponse {
  required string client_id = 1 [(nanopb).max_length = 4];
  required bytes payload = 2 [(nanopb).max_size = 14];
  required bytes auth_tag = 3 [(nanopb).max_size = 8];
}

message RemoteSignalResponsePayload {
  required uint32 code = 1;
  required bool success = 2;
  required bytes random = 3 [(nanopb).max_size = 8];
}
